src/MLIR/AST.hs

```diff
@@ -200,7 +206,13 @@ C.include "mlir-c/BuiltinTypes.h"
 C.include "mlir-c/BuiltinAttributes.h"
 
 instance FromAST Location Native.Location where
-  fromAST ctx _ UnknownLocation = Native.getUnknownLocation ctx
+  fromAST ctx _ loc = case loc of
+    UnknownLocation -> Native.getUnknownLocation ctx
+    FileLocation file line col -> do
```

Comment:
CI is complaining, because this pattern match is not exhaustive.

Reply:
Yes, will expand (had dog breaking a fence so got distracted :) ) - funnily `stack test` didn't complain here.

Reply:
Yeah I went with the usual setup of making CI use `-Werror`, but not using that in local builds (e.g. in case someone is using a newer compiler that emits warnings that our CI doesn't).